File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.java
class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        int mn = Arrays.stream(arr).min().getAsInt();
        int mx = Arrays.stream(arr).max().getAsInt();
        int n = arr.length;
        if(mx - mn == 0) return true;
        if((mx - mn) % (n-1) != 0) return false;
        int diff = (mx - mn) / (n-1);
        Set<Integer> s = new HashSet<>();
        for(int i=0; i<n; i++){
            if((arr[i] - mn) % diff != 0) return false;
            s.add(arr[i]);
        }
        return s.size() == n;
    }
}

File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        mn, mx = min(arr), max(arr)
        n = len(arr)
        if mx - mn == 0: return True
        if (mx - mn) % (n-1) != 0: return False
        diff = (mx - mn) // (n-1)
        s = set()
        for i in range(n):
            if (arr[i] - mn) % diff != 0: return False
            s.add(arr[i])
        return len(s) == n;